perfkitbenchmarker/linux_benchmarks/object_storage_service_benchmark.py

```diff
@@ -346,48 +355,105 @@ def _ProcessMultiStreamResults(raw_result, operation, sizes,
     sizes: the object sizes used in the benchmark, in bytes.
     results: a list to append Sample objects to.
     metadata: dict. Base sample metadata
-
   """
 
+  num_streams = FLAGS.object_storage_multistream_num_streams
+
   if metadata is None:
     metadata = {}
-
-  records = json.loads(raw_result)
-  metric_name = 'Multi-stream %s latency' % operation
-
-  metadata = metadata.copy()
-  metadata['num_streams'] = FLAGS.object_storage_multistream_num_streams
+  metadata['num_streams'] = num_streams
   metadata['objects_per_stream'] = (
       FLAGS.object_storage_multistream_objects_per_stream)
 
-  overall_metadata = metadata.copy()
+  records_json = json.loads(raw_result)
+  records = pd.DataFrame(records_json)
+
+  all_streams_start, all_streams_duration = analyze.AllStreamsInterval(
```

Comment:
I was looking for a way to make this more clear, and I ended up adding a class called Interval, which lets me condense the start time and duration of an interval into a single variable. It doesn't shorten the code much, but I think it does make it more clear what's going on. I'm still on the fence about whether it was worth doing or not - please let me know what you think.